src/core/instance/instance.cpp

```diff
@@ -329,6 +329,19 @@ Instance *Instance::InitMultiple(uint8_t aIdx)
     return instance;
 }
 
+Instance &Instance::Get(uint8_t aIdx)
+{
+    void *instance = gMultiInstanceRaw + aIdx * INSTANCE_SIZE_ALIGNED;
+    return *static_cast<Instance *>(instance);
+}
+
+uint8_t Instance::GetIdx(Instance *aInstance)
+{
+    uint8_t idx =
+        (reinterpret_cast<uint64_t>(aInstance) - reinterpret_cast<uint64_t>(gMultiInstanceRaw)) / INSTANCE_SIZE_ALIGNED;
```

Comment:
How about casting to `uint8_t *` (this way we get number of bytes between the two pointers and should be safer than `uint64_t` cast).
```suggestion
        (reinterpret_cast<uint8_t *>(aInstance) - reinterpret_cast<uint8_t *>(gMultiInstanceRaw)) / INSTANCE_SIZE_ALIGNED;
```

Reply:
Yes, it's a good idea. Thanks for your suggestion.